test/unit/pages/patientdata.test.js

```diff
@@ -1121,2206 +1246,2280 @@ describe('PatientData', function () {
+            const notAddingDataProps = _.assign({}, completedDataQueryProps, {
+              addingData: { inProgress: false },
+            });
 
-        const laterDateRange = [
-          moment(instance.state.lastProcessedDateTarget).add(1, 'days').toISOString(),
-          dateRange[1],
-        ];
+            const notFetchingDataProps = _.assign({}, completedDataQueryProps, {
+              fetchingPatientData: false,
+            });
 
-        instance.handleChartDateRangeUpdate(laterDateRange);
-        sinon.assert.callCount(instance.processData, 2);
-      });
+            const addingDataProps = _.assign({}, completedDataQueryProps, {
+              addingData: { inProgress: true },
+            });
 
-      it('should trigger processing if the chart scroll limit is reached and there is data to process', () => {
-        instance.handleChartDateRangeUpdate(dateRange);
-        sinon.assert.callCount(instance.fetchEarlierData, 0);
-        sinon.assert.callCount(instance.processData, 0);
+            const fetchingDataProps = _.assign({}, completedDataQueryProps, {
+              fetchingPatientData: true,
+            });
 
-        wrapper.setState({
-          lastDiabetesDatumProcessedIndex: 0,
-        });
+            beforeEach(() => {
+              // set props twice to ensure both this.props and nextProps set
+              wrapper.setProps({
+                ...notAddingDataProps,
+                ...notFetchingDataProps,
+              });
+            });
 
-        instance.handleChartDateRangeUpdate(dateRange);
-        sinon.assert.callCount(instance.processData, 1);
-      });
-    });
+            it('should hide the loader if data is not being fetched or added to worker', () => {
+              const hideLoadingSpy = sinon.spy(instance, 'hideLoading');
 
-    context('bgLog chart', () => {
-      beforeEach(() => {
-        setChartType('bgLog');
-      });
+              wrapper.setProps({
+                ...notAddingDataProps,
+                ...notFetchingDataProps,
+              });
 
-      it('should not trigger data fetching if all fetched data has not been processed', () => {
-        instance.handleChartDateRangeUpdate(dateRange);
-        sinon.assert.callCount(instance.fetchEarlierData, 0);
+              // this.props.addingData.inProgress: false, nextProps.addingData.inProgress: false
+              // this.props.fetchingPatientData: false, nextProps.fetchingPatientData: false
+              sinon.assert.callCount(hideLoadingSpy, 1);
+              hideLoadingSpy.resetHistory();
 
-        wrapper.setState({
-          lastDatumProcessedIndex: -1,
-        });
+              wrapper.setProps({
+                ...notAddingDataProps,
+                ...fetchingDataProps,
+              });
 
-        let newDateRange = [
-          moment(instance.props.fetchedPatientDataRange.start).subtract(1, 'milliseconds').toISOString(),
-          dateRange[1],
-        ];
+  
```

Comment:
Intentionally skipping?

Reply:
Yes.  Will be re-enabled with upcoming PDF integration PR